ares-cli: init at 3.1.1

Rucadi opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Ares is the main cli tool used by developers for interacting with LG WebOs Tv, documentation of this tool can be found here:  https://webostv.developer.lge.com/develop/tools/cli-introduction

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Misterio77 reviewed (changes requested):
Hey, thanks for the PR!

Looks good overall. Some feedback:

- Could you format it with `nixfmt-rfc-style`?
- ~As this package is not in repology, it would be nice to have a [`updateScript`](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript) set up.~ See next comment.
  - ~You can probably steal this from other `mkYarnPackage` drvs: [micropad](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/micropad/update.sh), [pomerium](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/http/pomerium/updater.sh), etc.~

Misterio77 commented on pkgs/by-name/ar/ares-cli/package.nix:
```diff
@@ -0,0 +1,31 @@
+    hash = "sha256-HhS49jg6PrWzXEX5P+NgcznMZw4yJyI0EYoCZ8y6UeU=";
+  };
+
+  yarnLock = ./yarn.lock;
+  doDist = false;
```
```suggestion
  postPatch = ''
    ln -s npm-shrinkwrap.json package-lock.json
  '';

  dontNpmBuild = true;
  npmDepsHash = "sha256-fTEYQY0ZhwYKnc7+EsgnnGovYJ8w+norq7oLnVrWNZg=";
```

Misterio77 replied:
`buildNpmPackage` works nicely with `npm-shrinkwrap`, just symlink it to `package-lock.json`.

Misterio77 commented on pkgs/by-name/ar/ares-cli/package.nix:
```diff
@@ -0,0 +1,31 @@
+{ lib
+, mkYarnPackage
```
```suggestion
, buildNpmPackage
```

Misterio77 commented on pkgs/by-name/ar/ares-cli/package.nix:
```diff
@@ -0,0 +1,31 @@
+, mkYarnPackage
+, fetchFromGitHub
+}:
+mkYarnPackage rec {
```
```suggestion
buildNpmPackage rec {
```

Misterio77 reviewed (changes requested):
Actually, sorry, one more (bigger) thing.

This is incurring IFD, I believe (that's why ofborg is failing). If you want to go through with `mkYarnPackage`, I believe you'll have to use `fetchYarnDeps` instead of `yarnLock` directly.

But I noticed upstream has a shrinkwrap, you should use `buildNpmPackage` with it instead of generating and vendoring a yarn lock. Here's how to.

PS: for `buildNpmPackage`, `nix-update-script` just works(tm). Simply add this:
```
  passthru.updateScript = nix-update-script { };
```

Rucadi commented:
> Hey, thanks for the PR!
> 
> Looks good overall. Some feedback:
> 
> * Could you format it with `nixfmt-rfc-style`?
> * ~As this package is not in repology, it would be nice to have a [`updateScript`](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript) set up.~ See next comment.
>   
>   * ~You can probably steal this from other `mkYarnPackage` drvs: [micropad](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/micropad/update.sh), [pomerium](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/http/pomerium/updater.sh), etc.~

Thank you very much for the review, to be honest I didn't quite get why this was failing, since using mkYarnPackage seemed the right way to do it.

I'll try to update the whole PR with your inputs :)

Rucadi committed:
ares-cli: init at 3.1.1

Co-authored-by: Gabriel Fontes <hi@m7.rs>

Updated the package to 3.1.1
Removed meta.platforms as this is set by buildNpmPackage

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

Rucadi commented:
I pushed the new version and added you as a co-author, your feedback was very appreciated.

It seems like you left the package in a much better way :) However, I wonder if the mkYarnPackage should be fixed? For me it seems like it shouldn't IFD.

Don't hesitate to give additional feedback if needed 😊

Misterio77 commented:
> However, I wonder if the mkYarnPackage should be fixed?

Hmm I think it uses IFD by default for convenience. So that people using it out of nixpkgs tree don't have to bump the deps hash all the time. By using `offlineCache = fetchYarnDeps`, you avoid IFD at the cost of it being a FOD (thus needing a hash).

> Your feedback was very appreciated

I'm glad I could help! :heart:

Misterio77 reviewed (approved):
LGTM :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1647

SuperSandro2000 commented on pkgs/by-name/ar/ares-cli/package.nix:
```diff
@@ -0,0 +1,39 @@
+    mainProgram = "ares";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ rucadi ];
+    platforms = lib.platforms.all;
```
```suggestion
```
buildNpmPackage limits that already to a smaller list of platforms

JohnRTitor reviewed (approved):
I have updated the package to 3.1.1
LGTM, builds for me
